frontend/src/ui/Input.tsx

```diff
@@ -50,6 +50,93 @@ export const TextArea = React.forwardRef<HTMLTextAreaElement, TextAreaProps>(
     ),
 );
 
+type InputWithCheckboxProps = {
+    checkboxChecked: boolean;
+    setCheckboxChecked: (newValue: boolean) => void;
+    label: string;
+    input: ReactNode;
+}
+
+/** Checkbox with a label to enable/disable an adjacent input */
+export const InputWithCheckbox: React.FC<InputWithCheckboxProps> = (
+    { checkboxChecked, setCheckboxChecked, label, input }
+) => <div css={{ display: "flex", flexDirection: "row", alignItems: "center" }}>
+    <input
+        type="checkbox"
+        checked={checkboxChecked}
+        onChange={() => setCheckboxChecked(!checkboxChecked)}
+        css={{ margin: "0 4px" }}
+    />
+    <label css={{ color: COLORS.neutral90, fontSize: 14 }}>
+        {label}
+    </label>
+    {input}
+</div>;
+
+type TimeInputProps = {
+    timestamp: string;
+    setTimestamp: (newTime: string) => void;
+    disabled: boolean;
+}
+
+/** A custom three-part input for time inputs split into hours, minutes and seconds */
+export const TimeInput: React.FC<TimeInputProps> = ({ timestamp, setTimestamp, disabled }) => {
+    const timeParts = (/(\d+h)?(\d+m)?(\d+s)?/).exec(timestamp)?.slice(1) ?? [];
```

Comment:
I would say that the state should not be the formatted string. The state should either be an array with 3 numbers or it should be one number (seconds). That's then converted to the time string where needed.